### PR TITLE
feat: [CCM-10934]: RBAC V2 implementation with resource groups

### PR DESCRIPTION
**Pull request by jyotiarora-harness**
### Summary

RBAC Resource group implementation on ccm perspective folders

#### Screenshots

Screenshot 2023-01-25 at 5 40 11 PM
Screenshot 2023-01-25 at 5 40 19 PM
Screenshot 2023-01-25 at 5 26 48 PM
Screenshot 2023-01-25 at 5 40 38 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-ce/components/ResourceGroupModals/PerspectiveResourceModalBody.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+          )
+        }
+      }
+    ],
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
+  )
+
+  if (foldersLoading) return <PageSpinner />
+  return !foldersLoading && foldersDataContent?.length ? (
```
Comment (rajarshee-harness) [inline src/modules/75-ce/components/ResourceGroupModals/PerspectiveResourceModalBody.tsx:66]:
```suggestion
  return foldersDataContent?.length ? (
```

Comment (jenil-harness) [inline src/modules/75-ce/components/ResourceGroupModals/PerspectiveResourceModalBody.tsx:66]:
```suggestion
  return foldersDataContent?.length ? (
```

File: src/modules/75-ce/components/ResourceGroupModals/PerspectiveResourceRenderer.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+  const filteredFoldersData = data.filter(item => item.uuid && identifiers.includes(item.uuid))
+
+  if (foldersLoading) return <PageSpinner />
+
+  return (
+    <TableV2
+      columns={[
+        {
+          id: 'name',
```
Comment (jenil-harness) [inline src/modules/75-ce/components/ResourceGroupModals/PerspectiveResourceRenderer.tsx:39]: columns don't have names? 
knit - can memoise

Comment (jyotiarora-harness) [reply src/modules/75-ce/components/ResourceGroupModals/PerspectiveResourceRenderer.tsx:39]: Name is not required in the Columns definition of react-table. Ref: https://react-table-v7.tanstack.com/docs/api/useTable#column-options
Adding the columns inside useMemo

File: src/modules/75-ce/components/ResourceGroupModals/PerspectiveResourceModalBody.tsx (hunk 1 above)
Comment (rishik30) [reply src/modules/75-ce/components/ResourceGroupModals/PerspectiveResourceModalBody.tsx]: +1

File: src/modules/75-ce/components/ResourceGroupModals/PerspectiveResourceRenderer.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  const { accountIdentifier } = resourceScope
+
+  const { data: foldersListResult, loading: foldersLoading } = useGetFolders({
+    queryParams: {
+      accountIdentifier: accountIdentifier
+    }
+  })
+  const data = defaultTo(foldersListResult?.data, [])
+
+  const filteredFoldersData = data.filter(item => item.uuid && identifiers.includes(item.uuid))
```
Comment (rishik30) [inline src/modules/75-ce/components/ResourceGroupModals/PerspectiveResourceRenderer.tsx:31]: nit: no need to store the folders data in variable and then filter it. We can combine the two operations in a single line.
```js
const filteredFoldersData = defaultTo(foldersListResult?.data, []).filter(item => item.uuid && identifiers.includes(item.uuid))
```

**Review by rishik30 — APPROVED** [review]

**Review by vkbansal-harness — APPROVED** [review]

**Pull request merged**